// ang-jsoneditor/esm5/lib.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { JsonEditorComponent } from './jsoneditor/jsoneditor.component';
var NgJsonEditorModule = /** @class */ (function () {
    function NgJsonEditorModule() {
    }
    /**
     * @return {?}
     */
    NgJsonEditorModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: NgJsonEditorModule,
            providers: []
        };
    };
    NgJsonEditorModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        JsonEditorComponent
                    ],
                    exports: [
                        JsonEditorComponent
                    ]
                },] },
    ];
    return NgJsonEditorModule;
}());
export { NgJsonEditorModule };
